ReactNativeClient/lib/services/rest/Api.js

```diff
@@ -448,10 +448,11 @@ class Api {
 			output.parent_id = folder.id;
 		}
 
-		if ('source_url' in requestNote) output.source_url = requestNote.source_url;
-		if ('author' in requestNote) output.author = requestNote.author;
+		if (requestNote.source_url) output.source_url = requestNote.source_url;
```

Comment:
Why did you change `'source_url' in requestNote` to `requestNote.source_url`?

AFAIK in case there is no source_url, there will now be an `undefined` error. I could be wrong though. In any case, I would follow the coding style and not change the style

Reply:
Yes as @tessus said, please revert this change. If the user provides an empty string, we should set the value to an empty string and not let it undefined.